cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/submit/MasterEnvironmentSparkSubmitter.java

```diff
@@ -141,6 +152,22 @@ protected void onCompleted(boolean succeeded) {
     } else {
       sparkExecutionService.shutdownNow();
     }
+    try {
+      if (sparkDriverWatcherThread instanceof AutoCloseable) {
```

Comment:
it's better for SparkDriverWatcher to extend AutoCloseable so we don't need to check here